disperser/meterer/offchain_store.go

```diff
@@ -0,0 +1,269 @@
+	reservationTableName string,
+	onDemandTableName string,
+	globalBinTableName string,
+	logger logging.Logger,
+) (*OffchainStore, error) {
+
+	dynamoClient, err := commondynamodb.NewClient(cfg, logger)
+	if err != nil {
+		return nil, err
+	}
+	if reservationTableName == "" || onDemandTableName == "" || globalBinTableName == "" {
+		return nil, fmt.Errorf("table names cannot be empty")
+	}
+
+	return &OffchainStore{
+		dynamoClient:         dynamoClient,
+		reservationTableName: reservationTableName,
+		onDemandTableName:    onDemandTableName,
+		globalBinTableName:   globalBinTableName,
+		logger:               logger,
+	}, nil
+}
+
+type ReservationBin struct {
+	AccountID string
+	BinIndex  uint64
+	BinUsage  uint32
+	UpdatedAt time.Time
+}
+
+type PaymentTuple struct {
+	CumulativePayment uint64
+	BlobSize          uint32
+}
+
+type GlobalBin struct {
+	BinIndex  uint64
+	BinUsage  uint64
+	UpdatedAt time.Time
+}
+
+func (s *OffchainStore) UpdateReservationBin(ctx context.Context, accountID string, binIndex uint64, size uint32) (uint32, error) {
+	key := map[string]types.AttributeValue{
+		"AccountID": &types.AttributeValueMemberS{Value: accountID},
+		"BinIndex":  &types.AttributeValueMemberN{Value: strconv.FormatUint(binIndex, 10)},
+	}
+
+	update := map[string]types.AttributeValue{
+		"BinUsage": &types.AttributeValueMemberN{Value: strconv.FormatUint(uint64(size), 10)},
+	}
+
+	res, err := s.dynamoClient.UpdateItemIncrement(ctx, s.reservationTableName, key, update)
+	if err != nil {
+		return 0, fmt.Errorf("failed to increment bin usage: %w", err)
+	}
+
+	binUsage, ok := res["BinUsage"]
+	if !ok {
+		return 0, errors.New("BinUsage is not present in the response")
+	}
+
+	binUsageAttr, ok := binUsage.(*types.AttributeValueMemberN)
+	if !ok {
+		return 0, fmt.Errorf("unexpected type for BinUsage: %T", binUsage)
+	}
+
+	binUsageValue, err := strconv.ParseUint(binUsageAttr.Value, 10, 32)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse BinUsage: %w", err)
+	}
+
+	return uint32(binUsageValue), nil
+}
+
+func (s *OffchainStore) UpdateGlobalBin(ctx context.Context, binIndex uint64, size uint32) (uint64, error) {
+	key := map[string]types.AttributeValue{
+		"BinIndex": &types.AttributeValueMemberN{Value: strconv.FormatUint(binIndex, 10)},
+	}
+
+	update := map[string]types.AttributeValue{
+		"BinUsage": &types.AttributeValueMemberN{Value: strconv.FormatUint(uint64(size), 10)},
+	}
+	res, err := s.dynamoClient.UpdateItemIncrement(ctx, s.globalBinTableName, key, update)
+	if err != nil {
+		return 0, err
+	}
+
+	binUsage, ok := res["BinUsage"]
+	if !ok {
+		return 0, nil
+	}
+
+	binUsageAttr, ok := binUsage.(*types.AttributeValueMemberN)
+	if !ok {
+		return 0, nil
+	}
+
+	binUsageValue, err := strconv.ParseUint(binUsageAttr.Value, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+
+	return binUsageValue, nil
+}
+
+func (s *OffchainStore) FindReservationBin(ctx context.Context, accountID string, binIndex uint64) (*ReservationBin, error) {
```

Comment:
Is this used?

Reply:
Nope atm, it was intended for an API, which i now realized I've yet to add